File: src/Library.Api/Domain/Loan.cs
namespace Library.Api.Domain;

public class Loan
{
    public long Id { get; set; }

    public long BookId { get; set; }

    public long MemberId { get; set; }

    public DateTimeOffset LoanDate { get; set; }

    public DateTimeOffset DueDate { get; set; }

    public DateTimeOffset? ReturnDate { get; set; }

    public bool IsReturned { get; set; }
}

File: src/BooksLibrary.WebApp/src/app/core/models/loan.ts
import { Book } from './book';

export interface LoanDetail {
  id: number;
  bookId: number;
  memberId: number;
  loanDate: string;
  dueDate: string;
  isReturned: boolean;
  isOverdue: boolean;
  book: Book | null;
}

export interface CreateLoanRequest {
  bookId: number;
  memberId: number;
}